src/ManagedServices/ManagedServices/Commands/NewAzureRmManagedServicesDefinition.cs

```diff
@@ -0,0 +1,125 @@
+﻿// ----------------------------------------------------------------------------------
+//
+// Copyright Microsoft Corporation
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ----------------------------------------------------------------------------------
+
+using Microsoft.Azure.Commands.ResourceManager.Common.ArgumentCompleters;
+using Microsoft.Azure.Management.ManagedServices.Models;
+using Microsoft.Azure.PowerShell.Cmdlets.ManagedServices.Models;
+using System;
+using System.Collections.Generic;
+using System.Management.Automation;
+using System.Text;
+
+namespace Microsoft.Azure.PowerShell.Cmdlets.ManagedServices.Commands
+{
+    [Cmdlet(
+    VerbsCommon.New,
+    Microsoft.Azure.Commands.ResourceManager.Common.AzureRMConstants.AzureRMPrefix + "ManagedServicesDefinition",
+    DefaultParameterSetName = DefaultParameterSet,
+    SupportsShouldProcess = true), OutputType(typeof(PSRegistrationDefinition))]
+    public class NewAzureRmManagedServicesDefinition : ManagedServicesCmdletBase
+    {
+        protected const string DefaultParameterSet = "Default";
+
+        [Parameter(ParameterSetName = DefaultParameterSet, Mandatory = false, HelpMessage = "The subscription scope.")]
+        [ScopeCompleter]
+        public string Scope { get; set; }
+
+        [Parameter(ParameterSetName = DefaultParameterSet, Mandatory = true, HelpMessage = "The name of the Registration Definition.")]
+        public string RegistrationDefinitionName { get; set; }
+
+        [Parameter(ParameterSetName = DefaultParameterSet, Mandatory = true, HelpMessage = "The Managedby Tenant Identifier.")]
+        [ValidatePattern("^({){0,1}[0-9a-fA-F]{8}-[0-9a-fA-F]{4}-[0-9a-fA-F]{4}-[0-9a-fA-F]{4}-[0-9a-fA-F]{12}(}){0,1}$")]
+        public string ManagedByTenantId { get; set; }
+
+        [Parameter(ParameterSetName = DefaultParameterSet, Mandatory = true, HelpMessage = "The Managedby Principal Identifier.")]
+        [ValidatePattern("^({){0,1}[0-9a-fA-F]{8}-[0-9a-fA-F]{4}-[0-9a-fA-F]{4}-[0-9a-fA-F]{4}-[0-9a-fA-F]{12}(}){0,1}$")]
```

Comment:
These validatepatterns don't rhow very good errors - ypou are much better off throwing your own message at runtime if the id is not a guid

Reply:
Done.